src/applications/simple-forms/21-4138/config/constants.js

```diff
@@ -1,4 +1,32 @@
 import React from 'react';
+import { useDispatch } from 'react-redux';
+import { unifyPPFormData } from '../helpers';
+
+export const PrimaryActionLink = ({ href = '/', children, onClick = null }) => (
+  <div className="action-bar-arrow" style={{ maxWidth: '75%' }}>
+    <div className="vads-u-background-color--primary vads-u-padding--1">
+      <a className="vads-c-action-link--white" href={href} onClick={onClick}>
+        {children}
+      </a>
+    </div>
+  </div>
+);
+
+const PrimaryActionLinkWithOnClick = ({ href = '/' }) => {
+  const dispatch = useDispatch();
+
+  const handlePriorityProcessingOnClick = e => {
+    e.preventDefault();
+    unifyPPFormData(dispatch);
```

Comment:
hm, this could potentially have an issue with timing. it will finish after `window.location.href = href;`, but i'm not sure what happens if it completes in the in-between state of page loads, since is a full page refresh. you could test by going to the Network tab in dev tools, and try different "throttling" speeds, and see if it still behaves appropriately.

Reply:
Hmm, good callout. It looks like not only timing but also the data needs a bit more modification in order to be saved when an existing PP form doesn't exist.